packages/experiments/src/components/Sidebar/Sidebar.types.tsx

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ */
+
+import { IButtonProps, IButtonStyles } from 'office-ui-fabric-react/lib/Button';
+import { IContextualMenuItem } from 'office-ui-fabric-react/lib/ContextualMenu';
+import { IStyle, ITheme } from 'office-ui-fabric-react/lib/Styling';
+import { IComponentAs } from 'office-ui-fabric-react/lib/Utilities';
+import { SidebarColors } from './Sidebar.styles';
+
+export interface ISidebar {
+  /**
+   * Toggles the sidebar state to put the sidebar in or out of collapsed mode
+   * @type {(boolean) => void}
+   */
+  toggleCollapsed: () => void;
+
+  /**
+   * Sets the sidebar in or out of collapsed mode
+   * @type {(boolean) => void}
+   */
+  setCollapsed: (newValue: boolean) => void;
+
+  /**
+   * Gets the current collapsed state of the sidebar.
+   * @type {() => boolean}
+   */
+  getCollapsed: () => boolean;
+}
+
+export interface ISidebarProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Optional callback to access the ISidebar interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   * @type {(component: ISidebar | null) => void}
+   */
+  componentRef?: (component: ISidebar | null) => void;
+
+  /**
+   * Whether the Sidebar is collapsible or not. If it is, a collapsible button
+   * is added to the top, and each element will render their onRenderCollapsed element.
+   * @type {boolean}
+   */
+  collapsible?: boolean;
+
+  /**
+   * Callback for when the collapse state is toggled.
+   * @type {() => void}
+   */
+  onCollapseChanged?: () => void;
```

Comment:
Thought for a future iteration: Would it be useful to pass the collapsed state as a param in this callback?